### PR TITLE
bug 1518784, replace raw ldap with OIDC auth, r=nemo

### PR DESCRIPTION
**Pull request by Pike**
We're keeping an option for standard model auth for local
development, behind OIDC_DISABLE.

Noteworthy, we're using non-default scopes to retrieve data from
auth0, adding `profile` to get first and last name.

### CODE REVIEW

File: lib/auth/tests.py [hunk 1]
```diff
@@ -4,341 +4,125 @@
...
+            backend = ElmoOIDCBackend()
+            claims = {
+                'given_name': 'Axel',
+                'family_name': 'Hecht',
+                'email': 'axel@example.com'
+            }
+            user = backend.create_user(claims)
+            update_mock.assert_called_once_with(user, claims)
+
+    def test_no_update(self):
```
Comment (johngian) [inline lib/auth/tests.py]: Another approach would be to also mock ``update_user`` and assert that its never called.

File: .hgignore [hunk 2]
```diff
@@ -1,9 +0,0 @@
-# .hgignore - List of filenames hg should ignore
```
[the remark this answers is not in the document]
Comment (Pike) [reply .hgignore]: Yes, it's a left-over from old-old days, and I had the options to maintain it with the changes I do to ignoring config files, or to remove it. Boom it goes.

File: lib/auth/tests.py (hunk 1 above)
Comment (Pike) [reply lib/auth/tests.py]: I'm not sure what you meant here. That said I'm doing a couple of changes to the test here to make it easier to read.

**Comment by johngian** [comment]
r+

**Comment by Pike** [comment]
Thanks for the review, trying on stage now (and finding some configuration nits).

**Pull request merged**